### DISCUSSION TITLE
Allow arbitrary expressions in the date range control

### DISCUSSION THREAD

**Discussion by KodrAus** [datalust/seq-tickets#1220] [Ideas]
The calendar picker lets us specify date time bounds for the query in `yyyy-mm-dd HH:MM:ss` form. It would be nice if we could also add arbitrary expressions in there that operate on the `@Timestamp` field:

- String literals like `2021-05-13 10:05:36` would be treated as `DateTime('2021-05-13 10:05:36')`
- Other expressions like `now() - 3d` would be treated as-is

We could surface that the calendar ranges operate on the `@Timestamp` field somehow so users can match up these expressions with what they might write in a query itself.

**Comment by schallm** [comment]
Would love to see this implemented.  Could also add more options at the bottom of the date picker.  Currently `Now` is the only option.  How about adding options like `Today`, `Yesterday`, `1 hour ago`.

**Comment by GWeidmann** [comment]
And please allow me to enter an partial time; if I enter something like this ...

![image](https://github.com/datalust/seq-tickets/assets/97590580/46a885a7-85c2-4098-8008-e063e4788cd0)

I don't want to be forced to also enter ":00:00" at the end
